src/dynamic_load_unix.cpp

```diff
@@ -45,16 +45,16 @@ DynamicallyLoadTracingLibrary(const char* shared_library,
       new DynamicLibraryHandleUnix{handle}};
 
   const auto make_tracer_factory =
-      reinterpret_cast<decltype(OpenTracingMakeTracerFactory)*>(
+      reinterpret_cast<OpenTracingMakeTracerFactoryType**>(
           dlsym(handle, "OpenTracingMakeTracerFactory"));
-  if (make_tracer_factory == nullptr) {
+  if (make_tracer_factory == nullptr || *make_tracer_factory == nullptr) {
     error_message = dlerror();
```

Comment:
If `*make_tracer_factory == nullptr`, won't this need a different error message since `dlerror` wouldn't return anything?

Reply:
indeed. you're totally right.